src/core/inlines.js

```diff
@@ -22,6 +22,72 @@ import { renderInlineCitation } from "./render-biblio";
 export const name = "core/inlines";
 export const rfc2119Usage = {};
 
+/**
+ * @param {string} matched
+ */
+function inlineRFC2119Matches(matched) {
+  const normalize = matched.split(/\s+/).join(" ");
+  const nodeElement = hyperHTML`<em class="rfc2119" title="${normalize}">${normalize}</em>`;
+  // remember which ones were used
+  rfc2119Usage[normalize] = true;
+  return /** @type {HTMLElement} */ nodeElement;
+}
+
+/**
+ * @param {string} matched
+ */
+function inlineXrefMatches(matched) {
+  // slices "{{{" at the beginning and "}}}" at the end
+  const ref = matched.slice(3, -3).trim();
+  return /** @type {HTMLElement} */ ref.startsWith("\\")
+    ? document.createTextNode(`{{{${ref.slice(1)}}}}`)
+    : idlStringToHtml(ref);
+}
+
+/**
+ * @param {string} matched
+ * @param {Text} txt
+ * @param {Object} conf
+ */
+function inlineBibrefMatches(matched, txt, conf) {
+  // slices "[[" at the start and "]]" at the end
+  const ref = matched.slice(2, -2);
+  if (ref.startsWith("\\")) {
+    return /** @type {HTMLElement[]} */ [
+      document.createTextNode(`[[${ref.slice(1)}]]`),
+    ];
+  } else {
+    const { type, illegal } = refTypeFromContext(ref, txt.parentNode);
+    const cite = renderInlineCitation(ref);
+    const cleanRef = ref.replace(/^(!|\?)/, "");
+    if (illegal && !conf.normativeReferences.has(cleanRef)) {
+      showInlineWarning(
+        cite.childNodes[1], // cite element
+        "Normative references in informative sections are not allowed. " +
+          `Remove '!' from the start of the reference \`[[!${ref}]]\``
+      );
+    }
+
+    if (type === "informative" && !illegal) {
+      conf.informativeReferences.add(cleanRef);
+    } else {
+      conf.normativeReferences.add(cleanRef);
+    }
+    return /** @type {HTMLElement[]} */ cite.childNodes;
+  }
+}
+
+/**
+ * @param {string} matched
+ * @param {Text} txt
+ * @param {Map<string, string>} abbrMap
+ */
+function inlineAbbrMatches(matched, txt, abbrMap) {
+  return /** @type {HTMLElement} */ txt.parentElement.tagName === "ABBR"
```

Comment:
```suggestion
  return txt.parentElement.tagName === "ABBR"
```

No need to manually cast here.

Reply:
(also that returns a boolean)

Reply:
@marcoscaceres It returns `Node`. (This is actually a ternary operation 😆)

Reply:
oh lol, oops.